lib/iris/plot.py

```diff
@@ -481,6 +482,50 @@ def _get_plot_objects(args):
     return u_object, v_object, u, v, args
 
 
+def _shift_plot_sections(u_object, u, v):
+    """
+    Shifts subsections of u by multiples of 360 degrees within ranges
+    defined by the points where the line should cross over the 0/360 degree
+    longitude boundary.
+
+    e.g. [ 300, 100, 200, 300, 100, 300 ] => [ 300, 460, 560, 660, 820, 660 ]
+
+    """
+    # Convert coordinates to true lat-lon
+    src_crs = u_object.coord_system.as_cartopy_crs() \
+        if u_object.coord_system is not None else ccrs.Geodetic()
+    tgt_crs = ccrs.Geodetic(globe=src_crs.globe)
+    tgt_proj = ccrs.PlateCarree(globe=src_crs.globe)
+
+    points = tgt_crs.transform_points(src_crs, u, v)
+    startpoints = points[:-1, :2]
+    endpoints = points[1:, :2]
+    proj_x, proj_y, _ = tgt_proj.transform_points(src_crs, u, v).T
+
+    # Calculate the inverse geodesic for each pair of points in turn, and
+    # convert the start point's azimuth into a vector in the source coordinate
+    # system.
+    # TODO: Figure out what the parameters of the geodesic should be
+    geodesic = Geodesic()
```

Comment:
@marqh What's stumped me is that it's possible for the globe (`ccrs.Globe`) to have none of the `semimajor_axis`, `semiminor_axis` or `inverse_flattening` attributes. A `Globe` will always have an `ellipse`, which defaults to `'WGS84'`. Is it possible to derive the other parameters from this?